sycl/include/sycl/handler.hpp

```diff
@@ -3307,7 +3322,7 @@ class __SYCL_EXPORT handler {
   std::vector<detail::ArgDesc> MAssociatedAccesors;
   /// Struct that encodes global size, local size, ...
   detail::NDRDescT MNDRDesc;
-  std::string MKernelName;
+  detail::string MKernelName;
```

Comment:
Every use of this seems to expect an `std::string`. Do we really need to change this data member, or maybe just a few places where it's getting assigned to?

Reply:
Yes, this private data member should cross the ABI boundary.
So I had to convert it to detail::string to overcome the ABI incompatibility issue.

Reply:
Also, have you tried changing callees accepting `MKernelName` instead of modifying the callers?

Reply:
I can't check everywhere at this moment.
That's why this PR is only the initial changes as the title of PR says.
We will fix as we encounter from the customer's usage case.

> Also, have you tried changing callees accepting MKernelName instead of modifying the callers?
I am not sure if I understand this question correctly, so please correct me if I misunderstood you.
MKernelName is defined in the headerfile, and the callees (cpp files) need this data member.
So, they accept MKernelName, but I can't hand over std::string since it is crossing the ABI boundary.
So, I changed MKernelName to detail::string and then the callee re-construct std::string in the CPP file.

Reply:
Do they really need `std::string`? Can they keep using `detail::string` or `detail::string_view`?

Reply:
Well.. That's called code pollution.
We (including @sergey-semenov) do not want to propagate `detail::string` deep on the CPP side.
`detail::string` and `detail::string_view` should be limited to be used when crossing the ABI boundary. They are not meant to replace `std::string` and `std::string_view`.

Reply:
I only partially agree with that. I think many of these interfaces currently accept `const std::string &` and they have no reason to ask for that and not for `std::string_view`. And if so, then everything can work automatically without us making ugly `#ifdef`s: https://godbolt.org/z/Wc8Kd14Y8.